client/src/views/Results.vue

```diff
@@ -82,17 +82,19 @@ export default defineComponent({
     },
     methods: {
         async onAddKeyword(event: { value: string }): Promise<void> {
-            this.terms.push(event.value);
```

Comment:
@micahh2 Not sure why creating a new array here is better than reusing the current reference? Do you have any specific reason for that?

Reply:
In general in JavaScript it's not considered a good idea to modify state, but instead generate new state. The JIT usually makes the runtime cost negligible (or even faster on common architectures because of less cache invalidation), and it generally avoids bugs like the one this PR is trying to address. The way the code worked previously, when we updated the terms array directly, we're also updating the VueJS router's reference to it (even though that's not reflected in the url). Of course, the other solution would be to clone the object before passing it to VueJS's router, but it is generally safer to eschew destructive operators entirely that to worry about who has what references to it.

Reply:
I totally understand. I've added some comments. :+1: